daemon/src/taker_cfd.rs

```diff
@@ -343,6 +317,40 @@ where
         }
         Ok(())
     }
+
+    async fn handle_propose_roll_over(&mut self, order_id: OrderId) -> Result<()> {
+        if self.current_pending_proposals.contains_key(&order_id) {
+            anyhow::bail!("An update for order id {} is already in progress", order_id)
+        }
+
+        let proposal = RollOverProposal {
+            order_id,
+            timestamp: SystemTime::now(),
+        };
+
+        self.current_pending_proposals.insert(
+            proposal.order_id,
+            UpdateCfdProposal::RollOverProposal {
+                proposal: proposal.clone(),
+                direction: SettlementKind::Outgoing,
+            },
+        );
+        self.send_pending_update_proposals()?;
+
+        // we are likely going to need this one
+        self.oracle_actor
```

Comment:
Shall we do this one earlier, so we don't insert pending things in case it fails? (it is `send`, so I suppose we know if it failed)

Reply:
I can only fail if the actor is down which it really should never be.

Reply:
I've heard that before :D